### PR TITLE
Add JRuby 9.2 to CI

### PR DESCRIPTION
**Pull request by marcotc**
This PR adds basic support for running the tracer tests against JRuby.

Currently, we only run tracer core tests: integrations are not tested.

### CODE REVIEW

File: ddtrace.gemspec [hunk 1]
```diff
@@ -51,8 +51,11 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'yard', '~> 0.9'
   spec.add_development_dependency 'webmock', '~> 2.0'
   spec.add_development_dependency 'builder'
-  spec.add_development_dependency 'ruby-prof'
-  spec.add_development_dependency 'sqlite3', '~> 1.3.6'
+  if RUBY_PLATFORM != 'java'
```
Comment (delner) [inline ddtrace.gemspec:54]: Not something we have to address here, but we should consider moving these "development dependencies" entirely out of the gemspec to the `Gemfile`, as they drive our test suite mostly. (A change that would be analogous to what we did for `pry-nav`.)

File: lib/ddtrace/runtime/metrics.rb [hunk 2]
```diff
@@ -55,6 +55,8 @@ def flush
       def gc_metrics
         Hash[
           GC.stat.map do |k, v|
+            next if v.is_a?(Hash) # TODO: JRuby supports additional nested metrics
```
Comment (delner) [inline lib/ddtrace/runtime/metrics.rb:58]: Will be interesting to expand on this later.

File: spec/support/platform_helpers.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module PlatformHelpers
```
Comment (delner) [inline spec/support/platform_helpers.rb:1]: I dig this.

File: ddtrace.gemspec (hunk 1 above)
Comment (marcotc) [reply ddtrace.gemspec:54]: Sounds like a nice way to clean up the gemspec file 👍

**Pull request merged**